src/sentry/static/sentry/app/components/createAlertButton.tsx

```diff
@@ -0,0 +1,266 @@
+              <li>
+                {tct(
+                  '[yAxis] is not supported by alerts just yet. Select a different metric below and try again.',
+                  {
+                    yAxis: <StyledCode>{eventView.getYAxis()}</StyledCode>,
+                  }
+                )}
+              </li>
+            )}
+          </StyledUnorderedList>
+        </React.Fragment>
+      )}
+      <StyledCloseButton
+        icon={<IconClose color="yellow400" size="sm" isCircled />}
+        aria-label={t('Close')}
+        size="zero"
+        onClick={onClose}
+        borderless
+      />
+    </StyledAlert>
+  );
+}
+
+type Props = React.ComponentProps<typeof Button> & {
+  className?: string;
+  projects: Project[];
+  /**
+   * Discover query used to create the alert
+   */
+  eventView: EventView;
+  organization: Organization;
+  /**
+   * Called when the eventView does not meet the requirements of alert rules
+   */
+  onIncompatibleQuery: (incompatibleQuery: React.ReactNode) => void;
+  /**
+   * Called when the alert close button is clicked
+   */
+  onClose: () => void;
+  /**
+   * Called when the user is redirected to the alert builder
+   */
+  onSuccess: () => void;
+};
+
+function incompatibleYAxis(eventView: EventView): boolean {
+  const column = explodeFieldString(eventView.getYAxis());
+  if (column.kind === 'field') {
+    return true;
+  }
+
+  const eventTypeMatch = eventView.query.match(/event\.type:(transaction|error)/);
+  if (!eventTypeMatch) {
+    return false;
+  }
+
+  const dataset = eventTypeMatch[1];
+  const yAxisConfig = dataset === 'error' ? errorFieldConfig : transactionFieldConfig;
+
+  const invalidFunction = !yAxisConfig.aggregations.includes(column.function[0]);
+  // Allow empty parameters
+  const invalidParameter = !['', ...yAxisConfig.fields].includes(column.function[1]);
+
+  return invalidFunction || invalidParameter;
+}
+
+/**
+ * Provide a button that can create an alert from an event view.
+ * Emits incompatible query issues on click
+ */
+function CreateAlertButton({
+  projects,
+  eventView,
+  organization,
+  onIncompatibleQuery,
+  onSuccess,
+  onClose,
+  ...buttonProps
+}: Props) {
+  // Must have exactly one project selected and not -1 (all projects)
+  const hasProjectError = eventView.project.length !== 1 || eventView.project[0] === -1;
+  // Must have one or zero environments
+  const hasEnvironmentError = eventView.environment.length > 1;
+  // Must have event.type of error or transaction
+  const hasEventTypeError =
+    !eventView.query.includes('event.type:error') &&
+    !eventView.query.includes('event.type:transaction');
+  // yAxis must be a function and enabled on alerts
+  const hasYAxisError = incompatibleYAxis(eventView);
+  const errors: IncompatibleQueryProperties = {
+    hasProjectError,
+    hasEnvironmentError,
+    hasEventTypeError,
+    hasYAxisError,
+  };
+  const project = projects.find(p => p.id === String(eventView.project[0]));
```

Comment:
I tend to like the `.toString()` method for this. Up to you though. I've also seen us use 

```
`${value}`
```